mlflow/gateway/app.py

```diff
@@ -137,10 +173,20 @@ async def get_route(route_name: str) -> Route:
         )
 
     @app.get(MLFLOW_GATEWAY_CRUD_ROUTE_BASE)
-    async def search_routes() -> SearchRoutesResponse:
-        # placeholder route listing functionality
-
-        return {"routes": list(app.dynamic_routes.values())}
+    async def search_routes(page_token: Union[str, None] = None) -> SearchRoutesResponse:
+        if page_token is not None:
+            start_idx = SearchRoutesToken.decode(page_token).index
+        else:
+            start_idx = 0
+
+        end_idx = start_idx + MLFLOW_GATEWAY_NUM_ROUTES_PER_SEARCH_RESULTS_PAGE
+        routes = list(app.dynamic_routes.values())
+        result = {"routes": routes[start_idx:end_idx]}
+        if len(routes[end_idx:]) > 0:
+            next_page_token = SearchRoutesToken(index=end_idx)
+            result["next_page_token"] = next_page_token.encode()
+
+        return result
```

Comment:
/ai what is the fastest way to extract the first N values from a dictionary in Python?

Reply:
@mlflow thanks!
@dbczumar Never mind this comment